File: LeetCode/322-Coin_Change.java
public class Coin_Change {

  public static void main(String[] args) {
    //int[] coins = {1, 2, 5};  int amount = 11;    // 3
    //int[] coins = {2};        int amount = 3;     // -1
    int[] coins = {2,3,4};    int amount = 7;

    Solution_1 sol_1 = new Solution_1();
    int res = sol_1.coinChange(coins, amount);
    System.out.println("\nres:" + res);
  }
}

File: LeetCode/322-Coin_Change.js
/*
322. Coin Change
*/

const coinChange = function(coins, amount) {
  coins.sort((a, b) => b-a);

  let count = 0;
  for (let i = 0; i < coins.length; i++) {
    count = count + Math.floor(amount / coins[i]);
    amount = amount % coins[i];

    if (amount == 0)
      return count;
  }
  return -1;
}

//---------- Testing ----------
let coins = [1, 2, 5];    let amount = 11;
let min = coinChange(coins, amount);
console.log("\nmin:", min);
